IOMC/Input/src/MCFileSource.cc

```diff
@@ -32,7 +33,8 @@ MCFileSource::MCFileSource(const ParameterSet & pset, InputSourceDescription con
   }  
   
   reader_->initialize(fileName);  
-  produces<HepMCProduct>();
+  produces<HepMCProduct>("generator");
```

Comment:
I modified the following stages as well in the code to take generic Input tags that can be configured  in python when running 
~~~python
process.genParticles.src= cms.InputTag("source","generator")
process.g4SimHits.HepMCProductLabel = cms.InputTag("source","generator")
process.g4SimHits.Generator.HepMCProductLabel = cms.InputTag("source","generator")
~~~

and as well the output commands needs to retain the gen information:
~~~python
process.AODSIMoutput.outputCommands.extend([
		'keep GenRunInfoProduct_*_*_*',
        	'keep GenLumiInfoProduct_*_*_*',
		'keep GenEventInfoProduct_*_*_*',
		])

process.MINIAODSIMoutput.outputcommands.extend([
       'keep GenRunInfoProduct_*_*_*',
       'keep GenLumiInfoProduct_*_*_*',
       'keep GenEventInfoProduct_*_*_*',
])
~~~

The choice to mark it with label 'generator' was for clarity since 'source' is not a very descriptive collection name. 
Anyhow, if put or not, the changes to InputTags are still needed.